Mobile Buy SDK/Mobile Buy SDK/Operations/BUYRequestOperation.m

```diff
@@ -169,14 +168,24 @@ - (NSURLSessionDataTask *)request:(NSURLRequest *)request completion:(BUYRequest
 			json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
 		}
 		
-		NSHTTPURLResponse *httpResponse = (id)response;
-		if (!httpResponse.successful && !error) {
-			error = [[NSError alloc] initWithDomain:kShopifyError code:httpResponse.statusCode userInfo:json];
+		NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+		if (!error) {
```

Comment:
This change is a bit confusing. Seems funny to seemingly create an error using a URL response. How about just passing in the status code?